Neos.Flow/Classes/Cache/CacheManager.php

```diff
@@ -352,15 +364,35 @@ protected function flushClassCachesByChangedFiles(array $changedFiles): void
         $modifiedAspectClassNamesWithUnderscores = [];
         $modifiedClassNamesWithUnderscores = [];
         foreach ($changedFiles as $pathAndFilename => $status) {
-            if (!file_exists($pathAndFilename)) {
-                continue;
+            $classNameWithUnderscores = null;
+            if ($status !== ChangeDetectionStrategyInterface::STATUS_DELETED) {
+                $fileContents = file_get_contents($pathAndFilename);
+                $className = (new PhpAnalyzer($fileContents))->extractFullyQualifiedClassName();
+                if ($className === null) {
+                    continue;
+                }
+                $classNameWithUnderscores = str_replace('\\', '_', $className);
+            } else {
```

Comment:
technically we can discuss if we always want to use the psr-4 reverse lookup instead of parsing the file via `PhpAnalyzer` ... but for that we would need to know how stable this logic is... for example regarding symlinks